### DISCUSSION TITLE
long lines in install_requires

### DISCUSSION THREAD

**Discussion by Gaasmann** [pypa/setuptools#3344] [Q&A] [answered]
Hello,

I have a dependency for my package with a lot of extras (like apackage[extra1, extra2, extra3, extra4, ...]).
I'm not sure if I can split that into several lines in setup.cfg. I tried the following and some variations of it but I can't find something that works.
```ini
[options]
install_requires = 
    package1
    package2
    package3[
        extra1, extra2, extra3, extra4,
        extra5, extra6, extra7, extra8
    ]
```
pip tells me the following when I try to install:

> pkg_resources.extern.packaging.requirements.InvalidRequirement: Parse error at "']'": Expected string_end

Is there a way to split my bunch of extras on several lines?

Thanks

Nicolas

**Comment by abravalheri** [comment, marked as answer]
Hi @Gaasmann , I don't think this is possible using `setup.cfg`. The `.ini` syntax is very limited...

Maybe you can try [`pyproject.toml `](https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html) and a some [multiline TOML strings](https://toml.io/en/v1.0.0#string) (I never tried myself, not sure if it works).

**Reply by Gaasmann** [reply]
Hi @abravalheri, I'll try with the pyproject.toml. That said, this seems to be experimental. Do you see any risk to production applications? Issues with repackaging/installing aren't a big deal in my case.
Thanks!

**Reply by abravalheri** [reply]
The most experimental part in `pyproject.toml` is the `[tool.setuptools]` table and its subtables.
Since the `[project]` table is specified in PEP 621, if your project just needs this table to operate, it should be OK.

**Reply by Gaasmann** [reply]
Ah ok, cool then. Thanks a lot!
